File: LeetCode/blind75/3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        count = set()
        l = 0

        longest = 0
        for r, c in enumerate(s):
            
            while c in count:
                count.remove(s[l])
                l += 1
            
            count.add(c)
            longest = max(longest, r-l + 1)

        return longest
            

        

File: LeetCode/blind75/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        std::unordered_set<int> set{};
        int res{};
        int l{};

        for(int r = 0; r < s.length(); ++r) {
            while(set.contains(s[r])) {
                set.erase(s[l]);
                ++l;
            }
            set.insert(s[r]);
            res = std::max(res, r-l+1);
        }
        return res;
    }
};
